esmf_regrid/_esmf_sdo.py

```diff
@@ -129,6 +129,17 @@ def __init__(
         """
         self.lons = lons
         self.lats = lats
+        londims = len(self.lons.shape)
+        assert len(self.lons.shape) == londims
+        latdims = len(self.lats.shape)
+        assert len(self.lats.shape) == latdims
+        assert londims == latdims
```

Comment:
Should we be raising Errors here? What is the preferred way of doing things in iris?  I don't usually use `assert` in library code, I would do something like:
```python
if londims != latdims:
    raise ValueError("Latitude and Longitude dimensions must be of the same order.")
```

Reply:
Yes, I think i have been using assert as a bit of a shortcut. I'll see what I can do about raising proper errors.